### PR TITLE
ENG-8907: Changes to make initiator stats work with internal connecti…

### PR DESCRIPTION
**Pull request by manjujames**
…on used by importers --
New methods in Connection to get name and id per caller (importer) of the internal connection.

This must be merged along with the pull request on pro side because of changes in Connection interface.

### CODE REVIEW

File: src/frontend/org/voltdb/InternalClientResponseAdapter.java [hunk 1]
```diff
@@ -441,6 +460,23 @@ public long connectionId() {
...
+    public long connectionId(long clientHandle) {
+        InternalCallback callback = m_callbacks.get(clientHandle);
+        if (callback==null) {
+            m_logger.warn("Could not find caller details for client handle " + clientHandle + ". Using internal adapter level connection id");
+            return connectionId();
+        }
+
+        Long internalId = m_internalConnectionIds.get(callback.getInternalContext().getName());
+        if (internalId==null) {
+            m_logger.warn("Could not find internal connection id for client handle " + clientHandle + ". Using internal adapter level connection id");
```
Comment (akhanzode) [inline src/frontend/org/voltdb/InternalClientResponseAdapter.java]: rate limit this or debug as they could flood if we hit this often?

File: src/frontend/org/voltdb/InternalClientResponseAdapter.java [hunk 2]
```diff
@@ -236,6 +240,10 @@ public boolean createTransaction(final InternalConnectionContext context,
         if (!m_partitionExecutor.containsKey(partition)) {
             m_partitionExecutor.putIfAbsent(partition, CoreUtils.getSingleThreadExecutor("InternalHandlerExecutor - " + partition));
         }
+
+        if (!m_internalConnectionIds.containsKey(context.getName())) {
```
Comment (akhanzode) [inline src/frontend/org/voltdb/InternalClientResponseAdapter.java]: You can move this inside Runnable.

File: src/frontend/org/voltdb/InternalClientResponseAdapter.java (hunk 1 above)
Comment (manjujames) [inline src/frontend/org/voltdb/InternalClientResponseAdapter.java]: Yes, that makes sense.

**Comment by manjujames** [comment]
Squash merged.

**Pull request closed without merging**